### PR TITLE
doc: clarify lifespanSeconds config behavior

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

Clarify lifespanSeconds attribute documentation in kafka instances configuration.
    
To align with the current behavior after recent long lived evaluation instances

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: config/kafka-instance-types-configuration.yaml [hunk 1]
```diff
@@ -44,7 +44,11 @@
 #   - minInSyncReplicas: Minimum number of in sync replicas
 #   - replicationFactor: Replication factor
 #   - supportedAZModes: a list of the supported AZ modes. The possible values are "single", "multi"
-#   - lifespanSeconds: The limit lifespan of the kafka instance in seconds. If not specified then the instance never expires
+#   - [optional] lifespanSeconds: The limit lifespan of the kafka instance in seconds.
+#                                 This attribute is applied at kafka creation time. Updates to this attribute
+#                                 will not impact the already existing kafka instances.
+#                                 If not specified then the instance never expires.
```
Comment (ziccardi) [inline config/kafka-instance-types-configuration.yaml:50]: This is not accurate in case of EVAL instances

Comment (miguelsorianod) [reply config/kafka-instance-types-configuration.yaml:50]: hmm right. This assumes the "instance never expires" phrase is in the context of lifespanSeconds.
Maybe a different wording would be:
If not specified then the instance never expires as a cause of the lifespanSeconds attribute

What do you think?

Comment (ziccardi) [reply config/kafka-instance-types-configuration.yaml:50]: Yeah, that would be better.

Comment (miguelsorianod) [reply config/kafka-instance-types-configuration.yaml]: :+1: updated

File: config/kafka-instance-types-configuration.yaml [hunk 2]
```diff
@@ -44,7 +44,12 @@
 #   - minInSyncReplicas: Minimum number of in sync replicas
 #   - replicationFactor: Replication factor
 #   - supportedAZModes: a list of the supported AZ modes. The possible values are "single", "multi"
-#   - lifespanSeconds: The limit lifespan of the kafka instance in seconds. If not specified then the instance never expires
+#   - [optional] lifespanSeconds: The limit lifespan of the kafka instance in seconds.
+#                                 This attribute is applied at kafka creation time. Updates to this attribute
```
Comment (JameelB) [inline config/kafka-instance-types-configuration.yaml:48]: Might be good to note what this applies to? to calculate the date of when a Kafka instance expires (marked for deprovision)?

Comment (miguelsorianod) [reply config/kafka-instance-types-configuration.yaml]: :+1: Updated.

**Review by ziccardi — APPROVED** [review]
LGTM

**Review by JameelB — APPROVED** [review]

**Pull request merged**